### PR TITLE
feat: add dependency filtering capabilities

### PR DESCRIPTION
**Pull request by stefanhoth**
## New Feature: Dependency Filtering

Adds dependency filtering capability to the libyear tool with a focused, CLI-only approach.

### 🤔 Motivation

I like the tool a lot. What was missing for internal metrics was a way to do calculations for specific package scopes. While I could have used a combination of overrides and manual calculation of the JSON output, I concluded adding top-level filtering maintains the full strength of the tooling.  
Now I can use the normal command and it directly outputs a summary of all of our internal libraries.

### ✨ Features

* **CLI Option**: `--include-only` for regex-based filtering
* **Regex Patterns**: Support for complex include patterns
* **Multiple Patterns**: Support for multiple `--include-only` flags
* **Comprehensive Testing**: Full test coverage for filtering functionality

### �� Implementation

* New `filterDependencies` utility function
* Updated CLI argument parsing
* Updated TypeScript types and interfaces
* Simplified approach focused on inclusion-only filtering

### 📚 Documentation

* New "Filtering" section in README
* Usage examples with `npx libyear`
* Updated CLI help text

### 🧪 Testing

* Unit tests for all filtering scenarios
* Edge case handling (invalid regex, empty patterns)
* Performance considerations

### �� Files Changed

* `src/filter.ts` - New filtering utility
* `src/filter.test.ts` - Comprehensive test suite
* `src/types.ts` - Updated types for filtering
* `src/cli/args.ts` - CLI argument parsing
* `src/cli/configuration.ts` - Configuration handling
* `src/cli/index.ts` - CLI integration
* `src/libyear.ts` - Core filtering integration
* `README.md` - Documentation and examples

### 🔍 Usage Examples

```bash
# Include only TypeScript-related packages
npx libyear --include-only "^@types/"

# Include multiple patterns
npx libyear --include-only "^@types/" --include-only "eslint"

# Include packages from a specific scope
npx libyear --include-only "^@myorg/"
```

### 🎯 Design Decision

This implementation focuses on inclusion-only filtering to avoid the risk of users indefinitely excluding dependencies from enforcement, while still addressing the use case of checking specific package scopes. The `--include-only` naming makes it clear this is the only filtering option available.

### CODE REVIEW

**Comment by jdanil** [comment]
Thanks for the contribution! I appreciate the thorough testing, doc updates, `--help` output updates, and support via both CLI options and the config file.

I have been hesitant to support filtering (see [Can I whitelist (or allowlist) dependencies from throwing errors?](https://github.com/jdanil/libyear#can-i-whitelist-or-allowlist-dependencies-from-throwing-errors) because I've worried that users might start filtering out dependencies they're blocked from updating, dependencies that become unmaintained, etc. which would eventually diminish the benefit of the check.

But I can see the value in using it to get metrics on a particular subset of dependencies (such as internal libraries in your example).

Let me give this a bit more thought over this week and I'll get back to you.

**Comment by stefanhoth** [comment]
@jdanil Thanks for your feedback. I didn't do much research into the history of my idea. I understand you want uphold the "true meaning" of the project. At the same time, with changes like mine your tool could serve additional use-cases that carry the idea of "libyear as a single, meaningful number" into spaces that did not apply it before.

I'll hold my breath for your decision 😉

**Comment by jdanil** [comment]
@stefanhoth how would you feel about updating the PR to just support the `--include` CLI option? I hate to see some of your good work being reverted, but I think this would help avoid the situation of users indefinitely excluding dependencies from enforcement, while also addressing your use case of only checking specific package scopes.

File: src/cli/index.ts [hunk 1]
```diff
@@ -58,40 +60,42 @@ export const cli = async (): Promise<void> => {
...
+        pulseIndividual: validateLimit(limit?.pulse?.individual),
+        releasesCollective: validateLimit(limit?.releases?.collective),
+        releasesIndividual: validateLimit(limit?.releases?.individual),
+        majorCollective: validateLimit(limit?.major?.collective),
+        majorIndividual: validateLimit(limit?.major?.individual),
+        minorCollective: validateLimit(limit?.minor?.collective),
+        minorIndividual: validateLimit(limit?.minor?.individual),
+        patchCollective: validateLimit(limit?.patch?.collective),
+        patchIndividual: validateLimit(limit?.patch?.individual),
+      };
```
Comment (stefanhoth) [inline src/cli/index.ts:96]: Why This Change Was Made:

- Better Code Organization: Configuration loading is now separate from processing
- Improved Readability: The complex destructuring and processing is now in separate steps
- Easier Maintenance: Each step is clear and can be modified independently
- Consistent with Filtering: The includeOnly parameter needed to be passed to libyear(), so configuration loading was moved up to make it available

**Comment by stefanhoth** [comment]
> @stefanhoth how would you feel about updating the PR to just support the `--include` CLI option? I hate to see some of your good work being reverted, but I think this would help avoid the situation of users indefinitely excluding dependencies from enforcement, while also addressing your use case of only checking specific package scopes.

@jdanil No worries, I think that's a fair compromise. Please have a look and let me know if this works for you. Thank you for the consideration!

**Comment by stefanhoth** [comment]
@jdanil Ping, just in case you missed the update above.
